File: frontend/src/types/project.ts
export interface Project {
  id: number;
  name: string;
  description: string | null;
  owner_id: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface ProjectCreate {
  name: string;
  description?: string;
}

export interface ProjectUpdate {
  name?: string;
  description?: string;
  is_active?: boolean;
}

export interface Environment {
  id: number;
  name: string;
  code: string;
  base_url: string;
  description: string | null;
  is_default: boolean;
  project_id: number;
}

export interface EnvironmentCreate {
  name: string;
  code: string;
  base_url: string;
  description?: string;
}

export interface EnvironmentUpdate {
  name?: string;
  code?: string;
  base_url?: string;
  description?: string;
}

File: backend/app/schemas/project.py
from typing import Optional, List, Dict, Any
from datetime import datetime
from pydantic import BaseModel

# --- Environment Schemas ---
class EnvironmentBase(BaseModel):
    name: str
    code: str
    base_url: str
    description: Optional[str] = None
    headers: Optional[Dict[str, Any]] = None
    is_default: bool = False

class EnvironmentCreate(EnvironmentBase):
    pass

class EnvironmentUpdate(EnvironmentBase):
    pass

class EnvironmentInDBBase(EnvironmentBase):
    id: int
    project_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class Environment(EnvironmentInDBBase):
    pass

# --- Project Schemas ---
class ProjectBase(BaseModel):
    name: str
    description: Optional[str] = None
    is_active: Optional[bool] = True

class ProjectCreate(ProjectBase):
    pass

class ProjectUpdate(ProjectBase):
    pass

class ProjectInDBBase(ProjectBase):
    id: int
    owner_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class Project(ProjectInDBBase):
    environments: List[Environment] = []
